test_util/launch_configs/README.md

```diff
@@ -0,0 +1,50 @@
+  * `cloud_cluster` will used hard-coded templates to deploy a cluster of machines in a shared network on the provider of choice (currently only supports AWS). `deployment_name` and `provider` must be set. See `cloud_cluster` below for options. Only `aws` provider is currently supported for this deploy option
+  * `onprem` refers to using the DC/OS installer (`dcos_generate_config.sh`) to deploy DC/OS onto a set of provided hosts. Those hosts can either be provisioned with keywords or by having previously created a `cloud_cluster` with dcos-launch. `installer_url` must be set as must `dcos_config` (see below)
+    * If `provider` is set to `cloud_cluster`, then the only other required field is `provider_info_path`, which points towards the cluster info JSON created by dcos-launch for which you want to install DC/OS
+    * If `provider` is not set to `cloud_cluster`, dcos-launch will use that provider to create the linux cluster and then immediately start deploying DC/OS to it (currently only supports `aws` provider). The linux cluster can be configured in the same manner as deploying a stand alone linux cluster (see `cloud_cluster` below), however the `instance_count` is omitted as it will be implied from `num_masters`, `num_public_agents`, and `num_private_agents`
+* `deployment_name`: The high-level name assigned in the provider for this resource allocation. In AWS, this is equivalent to stack name. In Azure, this is the resource group name, etc
+* `provider`: Can be `aws`, `cloud_cluster`, or `azure`
+  * If `aws` is set, then `provider_info` must be supplied with subfields for `region`, `access_key_id`, and `secret_access_key`
+  * If `azure` is set, then `provider_info` must be supplied with subfields for `location`, `tenant_id`, `subscription_id`, `client_id`, and `client_secret`
+  * If `cloud_cluster` is set, then that means you wish to use a pre-existing linux-cluster deployment that you previously launched. In this case, you do not specific a `provider_info` and instead use a `provider_info_path` that points to the cluster info JSON supplied by DC/OS Launch to provision the linux cluster
+
+### Potentially Optional Fields
+* `ssh`: In deployment types `template` and `cloud_cluster` this section can be entirely ommitted and the deployment will still occur. However, use of the pytest command will not be possible (this is irrelevant for the case of `cloud_cluster` where there is no purpose for pytest). In `onprem` deployment type, ssh capability is required to use the installer.
+  * `user`: User that will be able to SSH into the cluster and run test. Cannot be set with `deployment_type: cloud_cluster` as it will be inferred from the baked-in linux cluster template.
+  * `private_key_path`: path to the private key that will allow the above user to SSH. Cannot be set if `key_helper` is true (see below). Must be set with `deployment_type: onprem`
```

Comment:
How about flattening this to `ssh_user` and `ssh_private_key_path` at the top level? I think keeping the config layers shallow makes for an easier to grok interface.